dtfs-central-api/src/v1/controllers/portal/gef-facility/update-facility.controller.js

```diff
@@ -0,0 +1,32 @@
+const { ObjectId } = require('mongodb');
+const db = require('../../../../drivers/db-client');
+
+exports.updateFacility = async (id, updateBody) => {
+  try {
+    const facilitiesCollection = await db.getCollection('facilities');
+    const dealsCollection = await db.getCollection('deals');
+
+    let updatedFacility;
+
+    const existingFacility = await facilitiesCollection.findOne({ _id: ObjectId(id) });
+
+    if (existingFacility) {
+      updatedFacility = await facilitiesCollection.findOneAndUpdate({ _id: ObjectId(id) }, { $set: updateBody }, { returnOriginal: false });
+      if (updatedFacility) {
+        // update facilitiesUpdated timestamp in the deal
+        const dealUpdateObj = { facilitiesUpdated: new Date().valueOf() };
+
+        await dealsCollection.findOneAndUpdate(
+          { _id: { $eq: ObjectId(existingFacility.dealId) } },
+          { $set: dealUpdateObj },
+          { returnOriginal: false },
+        );
+      }
+    }
+
+    return updatedFacility;
```

Comment:
Bit strange it could return an empty variable here, not a show stopper

Reply:
if it is inside the if statement, it complains for linting about a "Expected to return a value at the end of async arrow function .eslintconsistent-return"